incubator/blockchain-client-openliberty/templates/default/README.md

```diff
@@ -0,0 +1,96 @@
+    FABRIC_CHANNEL=mychannel
+    FABRIC_CONTRACT=myasset
+    FABRIC_DEFAULT_IDENTITY=admin
+
+## WalletManagement
+
+Wallet(s) are a concept in the fabric SDK that contain identities for transacting on your fabric network. To remain consistent with the terminology we have created the a WalletManager class that contains code for easily onboarding fabric identities into the service.  The WalletManager is intensionally extendable so other wallet implementations can easily be made use of. The purpose of this component is house all the fabric identifies your service will want to transact with. The approach is entirely configuration driven, there are 2 modes of operation out of the box.
+
+`FABRIC_WALLET_PROFILE={"type":"IN_MEMORY"}`
+
+`FABRIC_WALLET_PROFILE={"type":"FILE_SYSTEM","options":{"path":"/project/user-app/src/main/java/application/Org1"}}`
+
+Creating your own Wallet backend is as simple as creating Wallet Implementation, adding configuration and adding a configuration type so the wallet manager can load the new implementation when configured.
+
+## API and Data Model
+
+The API Layer is the REST API definition layer for the micro service. This layer defines the set of input and output for each REST API endpoints the service exposes. In the template this component is simply AssetService.java. Looking at the class you will notice a set of open api annotations for each function. It is a best practice to use the Open API specification to document each endpoint, this enables an easy way for developers to test and also provides self service documentation to client consumers of the service.
+
+The Data Model layer are the Java Object that represent the set of inputs and output you wish to model. In the sample this component is simply Asset.java.
+
+## Identity Mapping
+
+Every service should be secure and provide a mechanism for authentication. The strategies you choose are beyond the scope of what the application template can provide, however we thought it important there be a minimal set of components that address the issue of mapping a service user to a user of an organization of a fabric network that the service user is acting on behalf. Here is the explanation of how the provider identity parts are meant to be use …
```

Comment:
Somewhere, I think we should add something like: "The template that is provided with the stack makes no assumptions on the specific authentication mechanism used by the client. It assumes that the client invoking the microservice that is based on the template is a trusted client, and that it will have performed some form of authentication. The trusted client will then forward an authentication token - such as JSON Web Token or a SAML assertion - to the microservice. It is the developer's responsibility to extract the appropriate claims from that token and map those claims to a Fabric identity. The template provided with the stack only provides the building blocks to implement this logic - but no specific implementation."

Reply:
Thanks for the writeup